diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index e3f79bfde5..1efd14d7d3 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -389,6 +389,18 @@ def _generate_posix_vars():
     if _PYTHON_BUILD:
         vars['BLDSHARED'] = vars['LDSHARED']
 
+    # This section patches the build vars to replace the static install
+    # prefix with one dynamically determined at runtime.
+    root = os.path.realpath(sys.executable)
+    root = os.path.dirname(root)
+    root = os.path.dirname(root)
+
+    existing_prefix = vars['prefix']
+    for k in vars.keys():
+        v = vars[k]
+        if isinstance(v, str) and existing_prefix in v:
+            vars[k] = v.replace(existing_prefix, root)
+
     # There's a chicken-and-egg situation on OS X with regards to the
     # _sysconfigdata module after the changes introduced by #15298:
     # get_config_vars() is called by get_platform() as part of the
